File: app/src/main/java/com/crossbowffs/quotelock/collections/provider/QuoteCollectionProvider.java
package com.crossbowffs.quotelock.collections.provider;

import android.content.Context;
import android.database.sqlite.SQLiteOpenHelper;

import com.crossbowffs.quotelock.provider.AutoContentProvider;

/**
 * @author Yubyf
 */
public class QuoteCollectionProvider extends AutoContentProvider {
    public QuoteCollectionProvider() {
        super(QuoteCollectionContract.AUTHORITY, new ProviderTable[] {
                new ProviderTable(
                        QuoteCollectionContract.Collections.TABLE,
                        QuoteCollectionContract.Collections.CONTENT_ITEM_TYPE,
                        QuoteCollectionContract.Collections.CONTENT_TYPE)
        });
    }

    @Override
    protected SQLiteOpenHelper createDatabaseHelper(Context context) {
        return new QuoteCollectionHelper(context);
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/collections/provider/QuoteCollectionProvider.kt
package com.crossbowffs.quotelock.collections.provider

import android.content.Context
import android.database.sqlite.SQLiteOpenHelper
import com.crossbowffs.quotelock.provider.AutoContentProvider

/**
 * @author Yubyf
 */
class QuoteCollectionProvider : AutoContentProvider(
    QuoteCollectionContract.AUTHORITY, arrayOf(
        ProviderTable(
            QuoteCollectionContract.Collections.TABLE,
            QuoteCollectionContract.Collections.CONTENT_ITEM_TYPE,
            QuoteCollectionContract.Collections.CONTENT_TYPE
        )
    )
) {
    override fun createDatabaseHelper(context: Context): SQLiteOpenHelper {
        return QuoteCollectionHelper(context)
    }
}